telegram/_utils/markup.py

```diff
@@ -31,14 +31,19 @@ class ``telegram.ReplyMarkup``.
 
 
 def check_keyboard_type(keyboard: object) -> bool:
-    """Checks if the keyboard provided is of the correct type - A list of lists.
+    """Checks if the keyboard provided is of the correct type - A sequence of sequences.
     Implicitly tested in the init-tests of `{Inline, Reply}KeyboardMarkup`
     """
     # string and bytes may actually be used for ReplyKeyboardMarkup in which case each button
     # would contain a single character. But that use case should be discouraged and we don't
     # allow it here.
     if not isinstance(keyboard, Sequence) or isinstance(keyboard, (str, bytes)):
```

Comment:
make this particular condition a new function, because we will be using it three times